rustls-libssl/src/cache.rs

```diff
@@ -315,6 +327,27 @@ impl ServerSessionStorage {
             self.flush_expired(TimeBase::now());
         }
     }
+
+    fn flush_oldest(items: &mut BTreeSet<Arc<SslSession>>) {
+        let mut oldest = None;
+
+        for item in items.iter() {
+            oldest = match oldest {
+                None => Some(item.clone()),
+                Some(oldest) => {
+                    if item.older_than(&oldest) {
+                        Some(item.clone())
+                    } else {
+                        Some(oldest)
+                    }
+                }
+            };
+        }
+
+        if let Some(oldest) = oldest {
+            items.take(&oldest);
+        }
+    }
```

Comment:
I _think_ that if the item `expiry_time` were made `pub` to this code this could be re-written to use `min_by_key()`, like:

```rust
    fn flush_oldest(items: &mut BTreeSet<Arc<SslSession>>) {
        let oldest = items.iter().min_by_key(|item| item.expiry_time.0);
        if let Some(oldest) = oldest.cloned()  {
            items.take(&oldest);
        }
    }
```

Possible I'm overlooking a reason to avoid doing this! I think the complexity is about the same but haven't reasoned it out very hard.

Reply:
Ah, nice!